backend/pkg/tls/tls_test_utils.go

```diff
@@ -0,0 +1,158 @@
+	keyPath := filepath.Join(tmpDir, "key.pem")
+	writeCertToPEMFile(t, certPath, serverCert)
+	writeKeyToPEMFile(t, keyPath, serverKey)
+
+	return &tlsTestSetup{
+		Logger:     logger,
+		TmpDir:     tmpDir,
+		CAPath:     caPath,
+		CACert:     caCert,
+		CAPrivKey:  caPrivKey,
+		ServerCert: serverCert,
+		CertPath:   certPath,
+		KeyPath:    keyPath,
+	}
+}
+
+// generateTestCACert creates a CA certificate for testing
+func generateTestCACert(t *testing.T) (*x509.Certificate, *rsa.PrivateKey) {
+	template := &x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		Subject: pkix.Name{
+			Organization: []string{"Test CA"},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(1 * time.Hour),
+		KeyUsage:              x509.KeyUsageCertSign | x509.KeyUsageCRLSign,
+		BasicConstraintsValid: true,
+		IsCA:                  true,
+		MaxPathLen:            1,
+	}
+
+	priv, err := rsa.GenerateKey(rand.Reader, 2048)
+	require.NoError(t, err, "Failed to generate private key")
+
+	derBytes, err := x509.CreateCertificate(rand.Reader, template, template, &priv.PublicKey, priv)
+	require.NoError(t, err, "Failed to create certificate")
+
+	cert, err := x509.ParseCertificate(derBytes)
+	require.NoError(t, err, "Failed to parse certificate")
+
+	return cert, priv
+}
+
+// generateServerCert creates a server certificate for testing
+func generateServerCert(t *testing.T, hostname string, caCert *x509.Certificate, caKey *rsa.PrivateKey) (*x509.Certificate, *rsa.PrivateKey) {
+	template := &x509.Certificate{
+		SerialNumber: big.NewInt(2),
+		Subject: pkix.Name{
+			Organization: []string{"Test Server"},
+		},
+		DNSNames:    []string{hostname},
+		NotBefore:   time.Now(),
+		NotAfter:    time.Now().Add(1 * time.Hour),
+		KeyUsage:    x509.KeyUsageDigitalSignature | x509.KeyUsageKeyEncipherment,
+		ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+	}
+
+	priv, err := rsa.GenerateKey(rand.Reader, 2048)
+	require.NoError(t, err, "Failed to generate private key")
+
+	derBytes, err := x509.CreateCertificate(rand.Reader, template, caCert, &priv.PublicKey, caKey)
+	require.NoError(t, err, "Failed to create certificate")
+
+	cert, err := x509.ParseCertificate(derBytes)
+	require.NoError(t, err, "Failed to parse certificate")
+
+	return cert, priv
+}
+
+// writeCertToFile writes a certificate to a file in PEM format
+func writeCertToFile(t *testing.T, path string, cert *x509.Certificate) {
+	certPEM := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: cert.Raw})
+	err := os.WriteFile(path, certPEM, 0o600)
+	require.NoError(t, err, "Failed to write certificate to file")
+}
+
+// writeCertToPEMFile writes a certificate to a file in PEM format
+func writeCertToPEMFile(t *testing.T, path string, cert *x509.Certificate) {
+	certPEM := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: cert.Raw})
+	err := os.WriteFile(path, certPEM, 0o600)
+	require.NoError(t, err, "Failed to write certificate to PEM file")
+}
```

Comment:
Why do we have 2 of these functions? They are exactly the same.

Reply:
Good call. thanks